indico/util/user.py

```diff
@@ -170,5 +192,24 @@ def principal_from_identifier(identifier, allow_groups=False, allow_external_use
         if event_role is None or event_role.event_id != event_id:
             raise ValueError('Invalid event role: {}'.format(event_role_id))
         return event_role
+    elif type_ == 'CategoryRole':
+        event = Event.get(event_id)
+        if event is None:
+            raise ValueError('Invalid event id: {}'.format(event_id))
+        if not allow_category_roles:
+            raise ValueError('Category roles are not allowed')
```

Comment:
I'd move this check up a bit (ie right before `event = Event.get(event_id)`) since getting the event is pointless if category roles are disabled. No harm in having it down here though, since the only case where you'd fail that condition is when sending nonsense data...